python/lsst/daf/butler/datastores/chainedDatastore.py

```diff
@@ -551,6 +551,10 @@ def remove(self, ref: DatasetRef) -> None:
         self.trash(ref, ignore_errors=False)
         self.emptyTrash(ignore_errors=False)
 
+    def forget(self, refs: Iterable[DatasetRef]) -> None:
+        for datastore in self.datastores:
```

Comment:
Maybe add:

    refs = list(refs)

before this line to avoid problems with true iterators?